### PR TITLE
Issue 21 serial interval methods

### PR DESCRIPTION
**Pull request by I-Bouros**
Created two new methods for the BranchProModel class:

- [x] a get_serial_intevals

- [x] an update_serial_intervals

Also added tests. Solves issue #21

### CODE REVIEW

File: branchpro/models.py [hunk 1]
```diff
@@ -42,3 +43,129 @@ def simulate(self, parameters, times):
...
+        at time t.
+    """
+
+    def __init__(self, initial_r, serial_interval):
+        super(BranchProModel, self).__init__()
+
+        if not isinstance(serial_interval, list):
+            raise TypeError('Serial interval values must be in a list format.')
+        if np.sum(serial_interval) <= 0:
+            raise ValueError('Sum of serial interval values must be > 0.')
```
Comment (Benjamin-Lee) [inline branchpro/models.py:80]:
```suggestion
            raise ValueError(f'Sum of serial interval values must be > 0 but it is {np.sum(serial_interval)}.')
```

File: branchpro/models.py [hunk 2]
```diff
@@ -42,3 +43,129 @@ def simulate(self, parameters, times):
...
+    simulate: return model output for specified parameters and times.
+    _normalised_daily_mean: (Private) returns the expected number of new cases
+        at time t.
+    """
+
+    def __init__(self, initial_r, serial_interval):
+        super(BranchProModel, self).__init__()
+
+        if not isinstance(serial_interval, list):
+            raise TypeError('Serial interval values must be in a list format.')
```
Comment (Benjamin-Lee) [inline branchpro/models.py:78]: Should this be able to take an np.array as well as a list?

File: branchpro/models.py [hunk 3]
```diff
@@ -42,3 +43,129 @@ def simulate(self, parameters, times):
...
+        if np.sum(serial_interval) <= 0:
+            raise ValueError('Sum of serial interval values must be > 0.')
+        if not isinstance(initial_r, (int, float)):
+            raise TypeError('Value of R must be integer or float.')
```
Comment (Benjamin-Lee) [inline branchpro/models.py:82]:
```suggestion
            raise TypeError(f'Value of R must be integer or float but got {type(initial_r)}.')
```

File: branchpro/models.py [hunk 4]
```diff
@@ -42,3 +43,129 @@ def simulate(self, parameters, times):
...
+
+        Parameters
+        ----------
+        new_serial_intervals
+            New unnormalised probability distribution of that the recipient
+            first displays symptoms s days after the infector first displays
+            symptoms.
+        """
+        if not isinstance(new_serial_intevals, list):
+            raise TypeError('Serial interval values must be in a list format.')
```
Comment (Benjamin-Lee) [inline branchpro/models.py:107]: Same as above. Why can't this be an np.array?

File: branchpro/models.py (hunk 3 above)
Comment (I-Bouros) [reply branchpro/models.py:94]: I see what you mean, and I do not disagree that it is useful, just isn't it a bit of an overkill? Not saying I cannot do it.

**Comment by DavAug** [comment]
I'll review this once #8 is merged.

**Review by DavAug — APPROVED** [review]
Looks very good!

**Pull request merged**